cEP-0020.md

```diff
@@ -0,0 +1,280 @@
+levels. But in case of Level-II and Level-III if a user is not able to acquire
+the required number of points after performing all the tasks, he will have to
+review more PRs to collect the required points.
+
+- **Badges** enable newcomers to follow their performance and compare to other
+newcomers.**After completing the level three successfully the newcomer will be a
+developer**. And he will be awarded a badge based on their performance in each
+level and the total number of points.
+
+These all are just basic ideas, the process and tasks in the levels can be
+changed according to the use case and implementaion process.
+
+A similar process can also be extended and applied to developers to achieve more
+number of badges or something more.
+
+## Implementation
+
+### Getting the data
+
+#### GitHub/GitLab
+
+For getting the data related to newcomers activity on GitHub/GitLab to the
+community, we will use [IGitt](https://gitlab.com/gitmate/open-source/IGitt),
+and since, our community repo is based on Django, we will use [igitt-django]
+(https://gitlab.com/gitmate/open-source/igitt-django) to do the initial setup of
+IGitt with Django. We will first implement `igitt-django` in the
+[webservices](https://gitlab.com/coala/landing/) and then import the useful data
+to the community repo with the use of our APIs.
+
+#### Gitter
+
+For getting the stats of newcomer's messages, we will use [Gitter API]
+(https://developer.gitter.im/docs/messages-resource) and import all the
+messages to newcomers or by newcomers in our webservices with some textual
+analysis for avoiding spam. And then we will import the final analysis of
+the messages to our community repo for tracking and gamification process.
+
+We will have different django models for each category like for GitHub/GitLab
+data we will have models named `PullRequest`, `Issue`, `Comment` and for Gitter
+data we will have models named `GoodMessage`, `BadMessage` and each of these
+models will be connected to a `User` model which will help us to track the
+activity of each user.
+
+### Tracking Activities
+
+After we have the data, we need to track some important activities of newcomers
+like:
+
+- Created Issues by a Newcomer
+- Created Pull Requests by a Newcomer
+- Number of Gitmate errors on the PR
+- CI status of the PR
+- Number of reviews done by a Newcomer
+- Number of comments in one review by a Newcomer
+- Comments on the Issues by a Newcomer
+- Activities on Gitter channels
+- Introduce `.coafile` in other projects
+
+We will have a dashboard showing these metrics of the user progress which will
+motivate newcomers to work faster to compete with others.
+
+#### Code Implementation
+
+Assuming that we are using [django-trackstats]
+(https://github.com/pennersr/django-trackstats) for tracking the statics of
+newcomers, then we will make different domains of similar set of metrics like
+for activities related to pull requests, we will have a domain named
+PullRequest. i.e:
+
+```
```

Comment:
It would be cool to replace all your first ` ``` ` with ` ```python ` so you get Python syntax highlighting to make your code stand out more :smile: 

(applies to all multi-line code snippets here)